File: app/src/main/java/com/tasomaniac/devdrawer/data/FilterDao.java
package com.tasomaniac.devdrawer.data;

import android.arch.persistence.room.*;
import io.reactivex.Flowable;
import io.reactivex.Single;

import java.util.List;

@Dao
public interface FilterDao {

  @Query("SELECT * FROM filter")
  Single<List<Filter>> allFilters();

  @Query("SELECT packageMatcher FROM filter WHERE appWidgetId = :appWidgetId")
  Flowable<List<String>> findFiltersByWidgetId(int appWidgetId);

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insertFilterSync(List<Filter> filter);

  @Query("DELETE FROM filter WHERE packageMatcher = :packageMatcher")
  void deleteFilterSync(String packageMatcher);

}

File: app/src/main/java/com/tasomaniac/devwidget/data/FilterDao.kt
package com.tasomaniac.devwidget.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Flowable
import io.reactivex.Single

@Dao
interface FilterDao {

    @Query("SELECT * FROM filter")
    fun allFilters(): Single<List<Filter>>

    @Query("SELECT packageMatcher FROM filter WHERE appWidgetId = :appWidgetId")
    fun findFiltersByWidgetId(appWidgetId: Int): Flowable<List<String>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFilterSync(filter: List<Filter>)

    @Query("DELETE FROM filter WHERE packageMatcher = :packageMatcher")
    fun deleteFilterSync(packageMatcher: String)
}
